lambdas/run/main.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	lambdaClient "github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/stripedpajamas/resl/models"
+)
+
+// RequestBody represents the model of the incoming request body
+type RequestBody struct {
+	TriggerID   string `json:"trigger_id"`
+	Text        string `json:"text"`
+	ResponseURL string `json:"response_url"`
+}
+
+var languageConfig models.LanguageConfig
+
+func processRequestBody(request events.APIGatewayProxyRequest) (RequestBody, error) {
+	fmt.Printf("Incoming request body: %s", request.Body)
+
+	var body RequestBody
+
+	if err := json.Unmarshal([]byte(request.Body), &body); err != nil {
+		fmt.Println(err)
+		return body, err
+	}
+
+	return body, nil
+}
+
+func getCodePayloadFromRequestBody(body RequestBody) ([]byte, error) {
+	var payload = parseText(body.Text)
+
+	if valid, err := validateLanguage(&payload); !valid {
+		return nil, err
+	}
+
+	parseCode(&payload)
+
+	return json.Marshal(payload)
+}
+
+func parseCode(payload *models.CodeProcessRequest) {
+	payload.Code = strings.ReplaceAll(payload.Code, "&amp;", "&")
+	payload.Code = strings.ReplaceAll(payload.Code, "&lt;", "&")
+	payload.Code = strings.ReplaceAll(payload.Code, "&gt;", "&")
+}
+
+func validateLanguage(payload *models.CodeProcessRequest) (bool, error) {
+	if _, ok := languageConfig[payload.Language]; !ok {
+		return false, errors.New("language not supported")
+	}
+
+	return true, nil
+}
+
+func parseText(text string) models.CodeProcessRequest {
+	text = strings.Trim(text, " ")
+	chars := []rune(text)
+	textLen := len(chars)
+
+	for idx, c := range text {
+		if c == ' ' {
+			return models.CodeProcessRequest{
+				Code:     string(chars[idx+1 : textLen]),
```

Comment:
can `text` be indexed/sliced directly? I thought strings already were rune slices

Reply:
updated